### ISSUE TITLE
Doctrine doesn't fill jsonb attributes of 2 or more classes who have fields with equal names

### ISSUE THREAD

**Issue by behamann** [boldtrn/JsonbBundle#8]
Hi, just noticed this bug.

2 classes with fields of same name, in one of them Doctrine couldn't bring the jsonb values:

```
/**
 * Entity.
 *
 * @ORM\Table(name="entity")
 * @ORM\Entity
 */
class Entity
{
    /**
     * @var array
     *
     * @ORM\Column(name="field", type="jsonb", nullable=true)
     */
    private $field;

    /**
     * Set field.
     *
     * @param jsonb $field
     *
     * @return Entity
     */
    public function setField($field)
    {
        $this->field= $field;

        return $this;
    }

    /**
     * Get field.
     *
     * @return jsonb
     */
    public function getField()
    {
        return $this->field;
    }
}

/**
 * SecondEntity.
 *
 * @ORM\Table(name="second_entity")
 * @ORM\Entity
 */
class SecondEntity
{
    /**
     * @var array
     *
     * @ORM\Column(name="field", type="jsonb", nullable=true)
     */
    private $field;

    /**
     * Set field.
     *
     * @param jsonb $field
     *
     * @return SecondEntity
     */
    public function setField($field)
    {
        $this->field= $field;

        return $this;
    }

    /**
     * Get field.
     *
     * @return jsonb
     */
    public function getField()
    {
        return $this->field;
    }
}
```

formTypes: 

```
class EntityType extends AbstractType
{
    /**
     * @param FormBuilderInterface $builder
     * @param array $options
     */
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        parent::buildForm($builder, $options);

        $builder
            ->add('field', 'collectionfield', array(
                'label' => 'entity.name',
                'label_attr' => [
                    'add_button' => 'entity.add',
                    'delete_button' => 'entity.delete',
                ]
            ))
        ;
    }

    /**
     * @param OptionsResolver $resolver
     */
    public function configureOptions(OptionsResolver $resolver)
    {
        $resolver->setDefaults(array(
            'data_class' => 'AppBundle\Entity\Entity'
        ));
    }

    /**
     * @return string
     */
    public function getName()
    {
        return 'appbundle_entity';
    }

}

class SecondEntityType extends AbstractType
{
    /**
     * @param FormBuilderInterface $builder
     * @param array $options
     */
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        parent::buildForm($builder, $options);

        $builder
            ->add('field', 'collectionfield', array(
                'label' => 'secondEntity.name',
                'label_attr' => [
                    'add_button' => 'secondEntity.add',
                    'delete_button' => 'secondEntity.delete',
                ]
            ))
        ;
    }

    /**
     * @param OptionsResolver $resolver
     */
    public function configureOptions(OptionsResolver $resolver)
    {
        $resolver->setDefaults(array(
            'data_class' => 'AppBundle\Entity\SecondEntity'
        ));
    }

    /**
     * @return string
     */
    public function getName()
    {
        return 'appbundle_second_entity';
    }

}
```

views is just the `{{ form_row(form.field) }}` on each of them

it saves on the DB, but if you make a simple `$entity = $em->getRepository('AppBundle:SecondEntity')->findOneBy(['id' => $id])`; the jsonb fields return [ ](empty array)

i think is a jsonb managing related bug, has I'm using your bundle I thought of posting it here

my solution at the moment was renaming the fields, and it went ok.

**Comment by boldtrn** [comment]
Hi @jbrunoxd 

Thanks for reporting the issue. Doctrine implemented the Jsonb datatype with Doctrine 2.5. If you can upgrade your current project to Doctrine 2.5 I would definitely recommend upgrading. 

Because of this I stopped the development of this Bundle. I wanted to update the Docs (will do this now).

Comments on your bug:
This sounds weird. I use this bundle in production use (will upgrade soon ;)). There I have the same jsonb field across different entities, with no issues. If you want, you could create a failing unit test, so I can take a look into this issue? I recommend upgrading Doctrine and see if the bug persists.

Btw: If you are querying for the id `$entity = $em->getRepository('AppBundle:SecondEntity')->findOneBy(['id' => $id])`, I recommend using the find method and not findBy.

**Comment by boldtrn** [comment]
I will close this issue since there was no update for several months
